### ISSUE TITLE
ZIP file could not be in the assets folder, you need to copy it on the filesystem

### ISSUE THREAD

**Issue by mysegfault** [MobileChromeApps/cordova-plugin-zip#56]
Hey,

I spent a LOT of time trying to figure out how to unzip a zip file included into my "assets" folder (Android)... because unzip() could never find my existing zip file... So the only solution is to copy the zip file from the assets into the device filesystem (i.e. cordova.file.externalCacheDirectory) with a 

fileEntry.copyTo(dirEntry, 'name.zip', function(fileEntryCopy)...

Then it's working fine! 

Please add a comment in the documentation about this requirement.

**Comment by srgpqt** [comment]
Also ran into this issue today.

**Comment by srgpqt** [comment]
The file existence check fails, even though the file does exist, and removing the check allows it to work.

The offending code:

```java
if (tempFile == null || !tempFile.exists()) {
    String errorMessage = "Zip file does not exist";
    callbackContext.error(errorMessage);
    Log.e(LOG_TAG, errorMessage);
    return;
}
```

**Comment by nguyentk90** [comment]
@mysegfault You're totally right. Thank you for that. We absolutely need to add this point to the document then everyone can know how to do it.
